core/src/collections.rs

```diff
@@ -0,0 +1,43 @@
+use crate::conn::MongoConnection;
+use crate::stmt::MongoStatement;
+use bson::{Array, Bson, Document};
+use mongodb::sync::Cursor;
+use std::error::Error;
+
+#[derive(Debug)]
+pub struct MongoCollections {
+    // The cursor on the result set.
+    resultset_cursor: Cursor<Document>,
+    // The current row the cursor points to.
+    current_row: Box<Array>,
+}
+
+// Statement related to a SQLColumns call.
+impl MongoCollections {
+    // Create a new MongoStatement to list tables with the given database (catalogs) and collection
+    // (tables) names filters.
+    // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
+    // timeout, the query must finish before the timeout or an error is returned.
+    pub fn list_tables(
+        client: &MongoConnection,
+        query_timeout: Option<i32>,
+        db_name_filter: &str,
+        collection_name_filter: &str,
+    ) -> Self {
+        unimplemented!()
+    }
+}
+
+impl MongoStatement for MongoCollections {
+    // Move the cursor to the next document and update the current row.
+    // Return true if moving was successful, false otherwise.
+    fn next(&mut self) -> Result<bool, Box<dyn Error>> {
```

Comment:
I'm not sure on the advantages of `dyn Error` versus just adding a pass through error using the `thiserror` package like we did in mongosql-rs. I'm not super opinionated either way since the main difference I see is performance, which isn't a big difference.

I suppose this may make it slightly harder to show errors, but `Error` implementers have to implement `Display`.

Reply:
@rychipman thoughts?

Reply:
Ah, this is because I thought that this way I would be able to just propagate the error from the Rust driver without having to do much and only create specific Core API error for issues happening in the Core API logic itself and still be able to return both.
I did not look into thiserror, so I'll check this and see what it does. It might totally enough for what I'm trying to achieve.

Reply:
I looked into thiserror and this is exactly what I wished for 🥳 I've changed the code to use it instead. For now, I only defined the "pass-through" error coming from the underlying Rust driver.